### DISCUSSION TITLE
Subscribe to single element changes in Form Array.

### DISCUSSION THREAD

**Discussion by olikari87** [react-hook-form/react-hook-form#6102] [Q&A]
Hey fam.

Start with thanking you guys for a great package and others for hopefully answering my question! 🙏 

Been reading the API Docs for Field Array and finding it hard to understand how to run an event handler for a **HTML select element** that is a in a Field Array. Usually one would just use onChange attribute but guess that's not the correct way to do it using React Hook Form?

Am I supposed to use **watch** for this situation? I don't want to watch the entire array, only individual fields.

Best regards.

**Comment by bluebill1049** [comment]
Hello :) thanks for the kind words. Do you want o try this instead? https://react-hook-form.com/api/usewatch

**Comment by bluebill1049** [comment]
here is an example: https://github.com/react-hook-form/react-hook-form/blob/master/app/src/useWatchUseFieldArrayNested.tsx

**Reply by olikari87** [reply]
May be I am super slow (and if so, sorry for that 🥴 ), but I think this example is not solving my case.

I guess this is where the subscription happens right, line 30?

`const value = useWatch({
  name: 'test',
  control,
});`

But this is watching changes in the whole array. I only want to subscribe to changes from individual elements inside the formArray, just like onChange attribute would give me an event from an individual element.

**My problem:** I have a field array, each containing an input and a select element. When one of the select element changes, I want to subscribe to that change and only that change.

Thanks again for taking the time BlueBill 🙏

**Reply by bluebill1049** [reply]
no worries :) 

what about this codesandbox: https://codesandbox.io/s/heuristic-rgb-n1flv?file=/src/App.js

you can subscribe to individual input.
